nixos/modules/services/x11/redshift.nix

```diff
@@ -82,18 +92,21 @@ in {
       '';
     };
 
-    extraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [];
-      example = [ "-v" "-m randr" ];
-      description = ''
-        Additional command-line arguments to pass to
-        <command>redshift</command>.
-      '';
-    };
   };
 
   config = mkIf cfg.enable {
+    services.redshift.configFile = mkDefault generatedConfig;
+
+    assertions = mkIf (cfg.configFile == generatedConfig) [ {
+        assertion = (cfg.settings ? redshift.dawn-time) == (cfg.settings ? redshift.dusk-time);
+        message = "Time of dawn and time of dusk must be provided together.";
+    } ];
+
+    services.redshift.settings.manual = {
+      lat = mkIf options.location.latitude.isDefined lcfg.latitude;
```

Comment:
I think this should be something like
```nix
let
  needsLocationProvider = ! cfg.settings ? redshift.dawn-time;
in mkIf (needsLocationProvider && lcfg.provider == "manual") {
  lat = lcfg.latitude;
  lon = lcfg.longitude;
}
```

Setting `lcfg == "manual"` should force the user to set `longitude` and `latitude` if it's needed. `isDefined` doesn't do that.